src/v2/Apps/Artist/Components/ArtistTopWorksRail/ArtistTopWorksRail.tsx

```diff
@@ -0,0 +1,150 @@
+import { Box, Flex, Sans } from "@artsy/palette"
+import React from "react"
+import { createFragmentContainer, graphql } from "react-relay"
+import { ArtistTopWorksRail_topRailArtworks } from "v2/__generated__/ArtistTopWorksRail_topRailArtworks.graphql"
+import { Carousel } from "v2/Components/Carousel"
+import FillwidthItem from "v2/Components/Artwork/FillwidthItem"
+import { Media } from "v2/Utils/Responsive"
+import { StyledLink } from "v2/Apps/Artist/Components/StyledLink"
+
+interface ArtistTopWorksRailProps {
+  topRailArtworks: ArtistTopWorksRail_topRailArtworks
+  onOverviewTab?: boolean
+}
+
+export const ArtistTopWorksRail: React.FC<ArtistTopWorksRailProps> = ({
+  topRailArtworks,
+  onOverviewTab,
+}) => {
+  const carouselHeight = 300
+  const artworks = topRailArtworks?.images?.edges ?? []
+  const handleViewWorksClick = event => {
+    return setTimeout(() => scrollToArtworkView(event), 500)
+  }
+
+  const scrollToArtworkView = event => {
+    const $element = document.getElementById("jump--artworkFilter")
+
+    if ($element) {
+      const { top } = getElementPosition($element)
+      window.scrollTo({
+        top: top,
+      })
+    }
+  }
+
+  const getElementPosition = $element => {
+    const rect = $element.getBoundingClientRect()
+    return {
+      left: rect.left + window.scrollX,
+      top: rect.top + window.scrollY,
+    }
+  }
+
+  return artworks.length > 0 ? (
+    <Flex mb="75px" flexDirection="column">
+      <Flex my={1} justifyContent="space-between">
+        <Sans size="4" color="black100">
+          Top Works
+        </Sans>
+        <Media at="xs">
+          {onOverviewTab ? (
+            <StyledLink
+              onClick={e => handleViewWorksClick(e)}
```

Comment:
Having an `onClick` on a `StyledLink` (or `RouterLink`) always confuses me. Clicking the link is already supposed to route you (or treat it as a regular `<a href=..` click), so basically navigates you. I get confused as to how the `onClick` interacts with that (does it override it? does it execute in addition to the navigation?)

It looks like this uses `setTimeout` to scroll to the filter (is that delay to account for being routed, and _then_ scrolled)? If so, clever! 👏

Reply:
I'm not sure but I would guess that the one of the use cases is to handle click tracking separately from routing?